app/controllers/application_controller.rb

```diff
@@ -1,10 +1,123 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions unless test?
+    set :session_secret, 'secret'
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    redirect to '/tweets' if logged_in?
+    erb :'users/create_user'
+  end
+
+  get '/login' do
+    redirect to '/tweets' if logged_in?
+    erb :'/users/login'
+  end
+
+  get '/tweets' do
+    redirect to '/login' unless logged_in?
+    @user = current_user
+    erb :'/tweets/tweets'
+  end
+
+  get '/tweets/new' do
+    redirect to '/login' unless logged_in?
+    erb :'/tweets/new'
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by(id: params[:id])
+      if @tweet.user == current_user
+        erb :'/tweets/edit_tweet'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  delete '/tweets/:id/delete' do
+    @tweet = Tweet.find(params[:id])
+    @tweet.delete if logged_in? && (@tweet.user == current_user)
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.